pydantic_ai_slim/pydantic_ai/result.py

```diff
@@ -126,6 +127,13 @@ def run_id(self) -> str:
         assert self._run_ctx.run_id is not None
         return self._run_ctx.run_id
 
+    @property
+    def metadata(self) -> dict[str, Any] | None:
+        """Metadata associated with this agent run, if configured."""
+        if self._metadata_getter is not None:
+            return self._metadata_getter()
```

Comment:
I understand that we do this because the metadata won't be available on `self._run_ctx` as it's only built at the end of the run, but I wonder if we can change that and simplify this.

Can we make sure the metadata is always available, just initially built off the original `RunContext` and then regenerated at the end of the run? Usually it'll just use `deps` anyway, which is available from the start.